packages/react-native/ReactCommon/react/renderer/components/view/BackgroundImagePropsConversions.h

```diff
@@ -0,0 +1,621 @@
+ */
+
+#pragma once
+
+#include <glog/logging.h>
+#include <react/debug/react_native_expect.h>
+#include <react/featureflags/ReactNativeFeatureFlags.h>
+#include <react/renderer/core/PropsParserContext.h>
+#include <react/renderer/core/RawProps.h>
+#include <react/renderer/graphics/BackgroundImage.h>
+#include <react/renderer/components/view/CSSConversions.h>
+#include <react/renderer/graphics/ColorStop.h>
+#include <react/renderer/graphics/LinearGradient.h>
+#include <react/renderer/graphics/RadialGradient.h>
+#include <react/renderer/graphics/ValueUnit.h>
+#include <react/renderer/css/CSSBackgroundImage.h>
+#include <react/renderer/css/CSSLengthUnit.h>
+#include <react/renderer/css/CSSPercentage.h>
+#include <react/renderer/css/CSSValueParser.h>
+#include <react/renderer/components/view/conversions.h>
+
+#include <optional>
+#include <string>
+#include <unordered_map>
+
+namespace facebook::react {
+inline void parseProcessedBackgroundImage(
+    const PropsParserContext& context,
+    const RawValue& value,
+    std::vector<BackgroundImage>& result) {
+  react_native_expect(value.hasType<std::vector<RawValue>>());
+  if (!value.hasType<std::vector<RawValue>>()) {
+    result = {};
+    return;
+  }
+
+  std::vector<BackgroundImage> backgroundImage{};
+  auto rawBackgroundImage = static_cast<std::vector<RawValue>>(value);
+  for (const auto& rawBackgroundImageValue : rawBackgroundImage) {
+    bool isMap = rawBackgroundImageValue
+                      .hasType<std::unordered_map<std::string, RawValue>>();
+    react_native_expect(isMap);
+    if (!isMap) {
+      result = {};
+      return;
+    }
+
+    auto rawBackgroundImageMap =
+        static_cast<std::unordered_map<std::string, RawValue>>(
+            rawBackgroundImageValue);
+
+    auto typeIt = rawBackgroundImageMap.find("type");
+    if (typeIt == rawBackgroundImageMap.end() ||
+        !typeIt->second.hasType<std::string>()) {
+      continue;
+    }
+
+    std::string type = (std::string)(typeIt->second);
+    std::vector<ColorStop> colorStops;
+    auto colorStopsIt = rawBackgroundImageMap.find("colorStops");
+
+    if (colorStopsIt != rawBackgroundImageMap.end() &&
+        colorStopsIt->second.hasType<std::vector<RawValue>>()) {
+      auto rawColorStops =
+          static_cast<std::vector<RawValue>>(colorStopsIt->second);
+
+      for (const auto& stop : rawColorStops) {
+        if (stop.hasType<std::unordered_map<std::string, RawValue>>()) {
+          auto stopMap =
+              static_cast<std::unordered_map<std::string, RawValue>>(stop);
+          auto positionIt = stopMap.find("position");
+          auto colorIt = stopMap.find("color");
+
+          if (positionIt != stopMap.end() && colorIt != stopMap.end()) {
+            ColorStop colorStop;
+            if (positionIt->second.hasValue()) {
+              auto valueUnit = toValueUnit(positionIt->second);
+              if (!valueUnit) {
+                result = {};
+                return;
+              }
```

Comment:
Is the expected behaviour that any intermediate invalid value cause the entire parse to fail?

Reply:
Yes, to match the web behaviour. e.g. linear-gradient(red 25xp, 42px) will draw an empty box on web. We did the [same](https://github.com/facebook/react-native/blob/5a4ed2d955fc5d45f4fbdd873a77419eb957d416/packages/react-native/Libraries/StyleSheet/processBackgroundImage.js#L652) for JS parser